app/common/notifications/notify.service.js

```diff
@@ -217,7 +218,52 @@ function Notify(_, $q, $rootScope, $translate, SliderService, ModalService, Demo
 
         return deferred.promise;
     }
+    function deleteWithInput (type, checkName) {
+        var scope = getScope();
+        var deferred = $q.defer();
+        scope.noequal = false;
+
+        const modalText = {
+            survey: {
```

Comment:
Not sure of this solution, maybe we should send the strings as arguments. I just found it so confusing with all the arguments for notifications.

Reply:
you mean instead of sending survey, send each individual string? 
I tend to like grouping as objects so I think this appraoch is great actually.

Reply:
if we start doing typescript you can even state the object structure more explicitly later on (not in this PR! ha) and that helps validate